platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/DataMigratable.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+/**
+ * Base contract for all migratables related to raw data. These are not mandatory and can be
```

Comment:
Same here:
> Interface that must be implemented by all bundles or features that need to export and/or import data during system migration. The data is not mandatory for the system to come up  (e.g., data stored in Solr) and doesn't affect the system's configuration, i.e., without this data, the new system will still have the same configuration as the original one.
> Any system related configuration and settings should be handled by a different service that implements the {@link ConfigurationMigratable} interface, not by implementors of this class.

We should also add the following disclaimer:
```
  * <p>
  * <b> This code is experimental. While this interface is functional and tested, it may change or be
  * removed in a future version of the library. </b>
  * </p>
```